pkg/providers/amifamily/bootstrap/eksbootstrap.go

```diff
@@ -65,86 +64,90 @@ func (e EKS) eksBootstrapScript() string {
 		caBundleArg = fmt.Sprintf("--b64-cluster-ca '%s'", *e.CABundle)
 	}
 	var userData bytes.Buffer
-	var kubeletExtraArgs strings.Builder
 	userData.WriteString("#!/bin/bash -xe\n")
 	userData.WriteString("exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1\n")
 	// Due to the way bootstrap.sh is written, parameters should not be passed to it with an equal sign
 	userData.WriteString(fmt.Sprintf("/etc/eks/bootstrap.sh '%s' --apiserver-endpoint '%s' %s", e.ClusterName, e.ClusterEndpoint, caBundleArg))
 
-	kubeletExtraArgs.WriteString(strings.Join([]string{e.nodeLabelArg(), e.nodeTaintArg()}, " "))
-
 	if e.isIPv6() {
 		userData.WriteString(" \\\n--ip-family ipv6")
 	}
+	if e.ContainerRuntime != "" {
+		userData.WriteString(fmt.Sprintf(" \\\n--container-runtime %s", e.ContainerRuntime))
+	}
+	if e.KubeletConfig != nil && len(e.KubeletConfig.ClusterDNS) > 0 {
+		userData.WriteString(fmt.Sprintf(" \\\n--dns-cluster-ip '%s'", e.KubeletConfig.ClusterDNS[0]))
+	}
 	if e.KubeletConfig != nil && e.KubeletConfig.MaxPods != nil {
 		userData.WriteString(" \\\n--use-max-pods false")
-		kubeletExtraArgs.WriteString(fmt.Sprintf(" --max-pods=%d", ptr.Int32Value(e.KubeletConfig.MaxPods)))
 	} else if !e.AWSENILimitedPodDensity {
 		userData.WriteString(" \\\n--use-max-pods false")
-		kubeletExtraArgs.WriteString(" --max-pods=110")
 	}
-	if e.KubeletConfig != nil && e.KubeletConfig.PodsPerCore != nil {
-		kubeletExtraArgs.WriteString(fmt.Sprintf(" --pods-per-core=%d", ptr.Int32Value(e.KubeletConfig.PodsPerCore)))
+	if args := e.kubeletExtraArgs(); len(args) > 0 {
+		userData.WriteString(fmt.Sprintf(" \\\n--kubelet-extra-args '%s'", strings.Join(args, " ")))
 	}
+	return userData.String()
+}
 
+func (e EKS) kubeletExtraArgs() (args []string) {
+	args = append(args, e.nodeLabelArg(), e.nodeTaintArg())
```

Comment:
can we check and bail if `e.KubeletConfig == nil` at the top  so we can get rid of all the reoccurring checks? (have to shift e.AWSENILimitedPodDensity into that check too, but still seems cleaner)